server.js

```diff
@@ -465,6 +374,369 @@ io.on('connection', (socket) => {
     }
 });
 
+//Add a bot to the game
+function addBot() {
+    numBots++;
+    botName = 'bot'+numBots;
+    shPlayers.push(botName);
+    if (readyPlayers.includes(botName))
+        readyPlayers.splice(readyPlayers.indexOf(botName), 1);
+    io.emit('sh-player-joined', {
+        name: botName,
+        num: shPlayers.length,
+        rp: readyPlayers
+    });
+    readyPlayers.push(botName);
+    io.emit('sh-ready-up', botName);
+    bots.push(botName);
+}
+
+//Bot nominates a chancellor
+function autoChancellor() {
+    if (bots.includes(presNom)){
+        console.log("A bot is president. Selecting a chancellor...");
+        botNominee = shPlayers[Math.floor(Math.random() * (shPlayers.length))];
+        while (undesirables.includes(botNominee) || presNom == botNominee) {
+            botNominee = shPlayers[Math.floor(Math.random() * (shPlayers.length))];
+        }
+        chanNom = botNominee;
+        io.emit('chancellor-nominated', {
+            c: chanNom,
+            p: presNom,
+            shp: shPlayers
+        });
+        autoVote();
+    }
+}
+
+//Bot votes
+function autoVote() {
+    if (fascists.includes(presNom)) {
+        fascistPres = true;
+    } else {
+        fascistPres = false;
+    }
+    if (fascists.includes(chanNom)) {
+        fascistChan = true;
+    } else {
+        fascistChan = false;
+    }
+    console.log("Bots voting...");
+    while (currentBot < bots.length) {
+        if (fascists.includes((bots[currentBot]))) {
+            if (fascistPres || fascistChan) {
+                    botYes();
+            } else {
+                    botNo();
+            }
+        } else {
+            if (Math.floor(Math.random()*2) == 1){
+                botYes();
+            } else {
+                botNo();
+            }
+        }
+        currentBot++;
+    }
+    currentBot = 0;
+}
```

Comment:
bots vote based on a simple algorithm, also prints "bots voting..." for debug purposes